### ISSUE TITLE
Version 6.1.6 of built_value_generator fails to run with the Dart SDK 2.0.0

### ISSUE THREAD

**Issue by pparvi** [google/built_value.dart#534]
### Test environment

With the simple project containing only `pubspec.yaml` file with the content;

```powershell
PS VSCode:\> type .\pubspec.yaml
name: built_test
environment:
  sdk: '>=2.0.0 <3.0.0'
dependencies:
  built_value: 6.1.6
dev_dependencies:
  build_runner: 1.1.1
  built_value_generator: 6.1.6
```

### Failure with the Dart SDKs before version 2.1.0-dev.5.0

When running `pub run build_runner build` command with the Dart 2.0.0 SDK, it fails with the following result:

````powershell
PS VSCode:\> pub run build_runner build
[INFO] Generating build script completed, took 627ms
[INFO] Creating build script snapshot... completed, took 6.5s
[SEVERE] Failed to snapshot build script .dart_tool/build/entrypoint/build.dart
PS VSCode:\> pub --version
Pub 2.0.0
PS VSCode:\> dart --version
Dart VM version: 2.0.0 (Fri Aug 3 10:53:23 2018 +0200) on "windows_x64"
````

When bisecting the failure, also Dart SDKs 2.1.0-dev.3.0 and 2.1.0-dev.4.0 gave the same result.

### Success with the Dart SDKs from 2.1.0-dev.5.0 onwards

With the Dart SDK 2.1.0-dev.5.0 same command completes succesfully:

````powershell
PS VSCode:\> pub run build_runner build
[INFO] Generating build script completed, took 378ms
[INFO] Creating build script snapshot... completed, took 8.4s
[INFO] Building new asset graph completed, took 523ms
[INFO] Checking for unexpected pre-existing outputs. completed, took 2ms
[INFO] Running build completed, took 7ms
[INFO] Caching finalized dependency graph completed, took 65ms
[INFO] Succeeded after 81ms with 0 outputs (0 actions)
PS VSCode:\> pub --version
Pub 2.1.0-dev.5.0
PS VSCode:\> dart --version
Dart VM version: 2.1.0-dev.5.0 (Wed Sep 19 19:15:19 2018 +0200) on "windows_x64"
````

Also with the Dart SDK 2.1.0-dev.6.0 command completed successfully.

### Result with the previous version

Reference result when using Dart SDK 2.0.0 with the version 6.1.5 of the built_value and built_value_generator:

```powershell
PS VSCode:\> type .\pubspec.yaml
name: built_test
environment:
  sdk: '>=2.0.0 <3.0.0'
dependencies:
  built_value: 6.1.5
dev_dependencies:
  build_runner: 1.1.1
  built_value_generator: 6.1.5
PS VSCode:\> pub get
<snip>
PS VSCode:\> pub run build_runner build
[INFO] Generating build script completed, took 628ms
[INFO] Creating build script snapshot... completed, took 9.8s
[INFO] Building new asset graph completed, took 643ms
[INFO] Checking for unexpected pre-existing outputs. completed, took 2ms
[INFO] Running build completed, took 12ms
[INFO] Caching finalized dependency graph completed, took 83ms
[INFO] Succeeded after 109ms with 0 outputs (0 actions)
PS VSCode:\> pub --version
Pub 2.0.0
PS VSCode:\> dart --version
Dart VM version: 2.0.0 (Fri Aug 3 10:53:23 2018 +0200) on "windows_x64"
```

**Comment by SocratesDz** [comment]
I'm experiencing the same output after an upgrade in my flutter environment. Sadly, I can't verify the same scenario (version 2.0.0 and version 2.1.0) right now. Here's a more verbose output if it helps.

```bash
>>> flutter packages pub run build_runner build -v
[INFO] Generating build script...
[INFO] Generating build script completed, took 323ms

[INFO] Creating build script snapshot......
[INFO] Creating build script snapshot... completed, took 10.7s

[SEVERE] Failed to snapshot build script .dart_tool/build/entrypoint/build.dart
pub finished with exit code 70

#0      throwToolExit (package:flutter_tools/src/base/common.dart:26:3)
#1      pubInteractively (package:flutter_tools/src/dart/pub.dart:186:5)
<asynchronous suspension>
#2      PackagesPassthroughCommand.runCommand (package:flutter_tools/src/commands/packages.dart:148:11)
<asynchronous suspension>
#3      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:383:18)
#4      _asyncThenWrapperHelper.<anonymous closure> (dart:async/runtime/libasync_patch.dart:77:64)
#5      _rootRunUnary (dart:async/zone.dart:1132:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#7      _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
#8      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
#9      Future._propagateToListeners (dart:async/future_impl.dart:671:32)
#10     Future._complete (dart:async/future_impl.dart:476:7)
#11     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#12     _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async/runtime/libasync_patch.dart:33:20)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
#16     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#17     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#18     _runPendingImmediateCallback (dart:isolate/runtime/libisolate_patch.dart:115:13)
#19     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:172:5)
>>> flutter packages pub --version
Flutter 0.10.2 • channel beta • https://github.com/flutter/flutter.git
Framework • revision d8cbb80206 (2 weeks ago) • 2018-10-26 01:30:21 -0400
Engine • revision 6c2ade9fa2
Tools • Dart 2.1.0-dev.8.0.flutter-bf26f760b1
```

**Comment by davidmorgan** [comment]
https://github.com/dart-lang/sdk/issues/35158

**Comment by pparvi** [comment]
Issue fixed with the release of analyzer 0.33.3+1.
